### PR TITLE
Small tweaks to Bare OS Cluster creation

### PR DESCRIPTION
**Pull request by brennancheung**
* Make Virtual IP conditionally required when number of master nodes > 1
* Allow user to specify multiple Metal LB IP pool ranges
* Change network interface picklist to only use interface name (not IP)

### CODE REVIEW

File: src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/VipInterfaceChooser.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { pathStrOr } from 'utils/fp'
+import { uniq } from 'ramda'
+
+const VipInterfaceChooser = forwardRef(({
+  masterNodes, hasError, errorMessage, ...rest
+}, ref) => {
+  const [nodes, loading] = useDataLoader(loadNodes)
+  const masters = nodes.filter(node => masterNodes.includes(node.uuid))
+
+  const options = uniq(masters.map(node => {
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/VipInterfaceChooser.js]: This looks a bit expensive, `useMemo` perhaps?

File: src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/AddBareOsClusterPage.js [hunk 2]
```diff
@@ -31,7 +32,7 @@ const runtimeConfigOptions = [
 ]
 
 const AddBareOsClusterPage = () => {
-  const { params } = useParams()
+  const { params, getParamsUpdater } = useParams()
   const { history } = useReactRouter()
   const onComplete = success => {
     if (!success) { return console.error('Unable to create cluster') }
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/AddBareOsClusterPage.js]: You don't need to do this, errors are already shown at console and a Toast notification will pop up too

File: src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/VipInterfaceChooser.js (hunk 1 above)
Comment (brennancheung) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/VipInterfaceChooser.js]: It's not expensive at all.  Typical scenario is between 2-5 nodes.  Not worth the added code complexity.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/AddBareOsClusterPage.js (hunk 2 above)
Comment (brennancheung) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/AddBareOsClusterPage.js]: I'm confused, your last comment on this PR suggested I use the `success` parameter.

Comment (xantorres) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/bareos/AddBareOsClusterPage.js]: It makes sense in some cases, eg to decide if you redirect the user to the list screen, or stay in there has been an error, so the user can fix any mistakes and try again

**Pull request merged**